src/lib/apiTypesV2/Entities.cpp

```diff
@@ -0,0 +1,136 @@
+* License, or (at your option) any later version.
+*
+* Orion Context Broker is distributed in the hope that it will be useful,
+* but WITHOUT ANY WARRANTY; without even the implied warranty of
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU Affero
+* General Public License for more details.
+*
+* You should have received a copy of the GNU Affero General Public License
+* along with Orion Context Broker. If not, see http://www.gnu.org/licenses/.
+*
+* For those usages not covered by this license please contact with
+* iot_support at tid dot es
+*
+* Author: Ken Zangelin
+*/
+#include <string>
+#include <vector>
+
+#include "ngsi10/QueryContextResponse.h"
+#include "apiTypesV2/Entities.h"
+
+
+
+/* ****************************************************************************
+*
+* Entities::Entities - 
+*/
+Entities::Entities()
+{
+  errorCode.fill(SccOk);
+}
+
+
+
+/* ****************************************************************************
+*
+* Entities::~Entities - 
+*/
+Entities::~Entities()
+{
+  release();
+}
+
+
+
+/* ****************************************************************************
+*
+* Entities::render - 
+*
+* If no error reported in errorCode, render the vector of entities.
+* Otherwise, render the errorCode.
+*/
+std::string Entities::render(ConnectionInfo* ciP, RequestType requestType)
+{
+  if ((errorCode.code == SccOk) || (errorCode.code == SccNone))
+  {
+    return vec.render(ciP, requestType, false);
+  }
+
+  return errorCode.renderV2();
+} 
+
+
+
+/* ****************************************************************************
+*
+* Entities::check - 
+*
+* NOTE
+*   The 'check' method is normally only used to check that incoming payload is correct.
+*   For now (at least), the Entities type is only used as outgoing payload ...
+*/
+std::string Entities::check(ConnectionInfo* ciP, RequestType requestType)
+{
+  return vec.check(ciP, requestType);
+}
+
+
+
+/* ****************************************************************************
+*
+* Entities::present - 
+*/
+void Entities::present(const std::string& indent)
+{
+  LM_F(("%s%d Entities:", indent.c_str(), vec.size()));
+  vec.present(indent + "  ");
+}
+
+
+
+/* ****************************************************************************
+*
+* Entities::release - 
+*/
+void Entities::release(void)
+{
+  vec.release();
+}
+
+
+
+/* ****************************************************************************
+*
+* Entities::fill - 
+*/
+void Entities::fill(QueryContextResponse* qcrsP)
+{
+  if (qcrsP->errorCode.code != SccOk)
+  {
+    errorCode.fill(qcrsP->errorCode);
+    return;
+  }
+
+  for (unsigned int ix = 0; ix < qcrsP->contextElementResponseVector.size(); ++ix)
+  {
+    ContextElement* ceP = &qcrsP->contextElementResponseVector[ix]->contextElement;
+    Entity*         eP  = new Entity();
+
+    eP->id        = ceP->entityId.id;
+    eP->type      = ceP->entityId.type;
+    eP->isPattern = ceP->entityId.isPattern;
```

Comment:
The idea is for this method to be generic.
In the case of rendering a GET, we don't want isPattern, but in the case of filling the Entity after parsing payload from a POST, isPattern might have to be there.